### PR TITLE
feat(RequestHandler): add `modelName` to `handleAndLogRequestError` params

### LINKED ISSUE #20724: Add model info to PrismaClientKnownRequestError
## Problem

I have global exception filter for NestJS, and it's catcher only have error instance of `PrismaClientKnownRequestError`. This execption filter needs somehow to know, what model is related to this error, e.g. to provide custom error message, related to model name.

Real example - trying to create user, but user with provided email exists in database. Exception filter uses error code (`P2002`) to detect this is unique constraint error or something else, but it still doesn't know what model it's related for.

## Suggested solution

Add model name to error metadata, e.g.:
```js
{
  meta: {
    model: 'User',
    target: [
      'email'
    ]
  }
}
```
## Alternatives

Imho best approach will be attaching custom metadata for any query, e.g.:
```js
prismaClient.user.create({
  data: {
    ...
  },
  customMetadata: {
    ... // any data engineer wants to attach
  },
});
```
Idk how it's possible and/or type safe, but this grants far more flexibility than current usage.

### PR DESCRIPTION
**Pull request by AikoRamalho**
Closes [#20724](https://github.com/prisma/prisma/issues/20724)
This PR enhances error handling in the codebase by including the modelName parameter in the error handling process. This modification ensures better contextual information is available when handling and logging errors during a request.

Changes
Added modelName to the parameters passed in the catch block of the try-catch statement.
Updated the HandleErrorParams interface and modified the handleRequestError method to include the modelName parameter.
Motivation
When an error occurs during a request, having the modelName information at hand can significantly improve the debugging and error resolution process. This update provides more comprehensive details about the context in which an error occurred, enhancing the overall maintainability and troubleshooting capabilities of the code.

### CODE REVIEW

**Comment by AikoRamalho** [comment]
Closes [#20724](https://github.com/prisma/prisma/issues/20724)

**Comment by AikoRamalho** [comment]
@SevInf Your thoughts here would be very much appreciated! 😀

File: packages/client/src/runtime/RequestHandler.ts [hunk 1]
```diff
@@ -213,7 +214,7 @@ export class RequestHandler {
       throw new PrismaClientKnownRequestError(message, {
         code: error.code,
         clientVersion: this.client._clientVersion,
-        meta: error.meta,
+        meta: { modelName, ...error.meta },
```
Comment (SevInf) [inline packages/client/src/runtime/RequestHandler.ts:217]: `modelName` is actually optional. Some top-level operations like `$queryRaw` don't have it. Could you ensure that `modelName: undefined` does not get added to `meta` in this case?

**Comment by SevInf** [comment]
Thank you @AikoRamalho, overall looks good, left one small comment.
Could you also add a test for this? You can find instructions [here](https://github.com/prisma/prisma/blob/main/TESTING.md#prisma-client-functional-tests). It's ok if you don't have time for that, just let me know and I take it from here.

**Comment by AikoRamalho** [comment]
> an find instructions [here](https://github.com/prisma/prisma/blob/main/TESTING.md#prisma-client-functional-tests). It's ok if you don't have time for that, just let me know and I take it from here.

Working on it!

**Comment by AikoRamalho** [comment]
Hey @janpio!

Your thoughts here would be very much appreciated!

thank you!!!

**Review by SevInf — APPROVED** [review]
@AikoRamalho looks good, thank you and congratulations on the first contribution.

**Pull request merged**